app/src/main/kotlin/com/wire/android/workmanager/worker/SingleUserMigrationWorker.kt

```diff
@@ -0,0 +1,116 @@
+import com.wire.kalium.logic.data.id.QualifiedIdMapperImpl
+import com.wire.kalium.logic.data.user.UserId
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.filterNotNull
+import kotlinx.coroutines.flow.map
+import java.util.concurrent.TimeUnit
+
+@HiltWorker
+class SingleUserMigrationWorker @AssistedInject constructor(
+    @Assisted appContext: Context,
+    @Assisted workerParams: WorkerParameters,
+    private val migrationManager: MigrationManager,
+    private val notificationChannelsManager: NotificationChannelsManager
+) : CoroutineWorker(appContext, workerParams) {
+
+    override suspend fun doWork(): Result = coroutineScope {
+        val userId = inputData.getString(USER_ID_INPUT_DATA)?.let { userId ->
+            QualifiedIdMapperImpl(null).fromStringToQualifiedID(userId)
+        } ?: return@coroutineScope Result.failure()
+
+        when (migrationManager.migrateSingleUser(userId, this) { setProgress(it.type.toData()) }) {
+            is MigrationData.Result.Success -> Result.success()
+            is MigrationData.Result.Failure -> Result.retry()
+        }
+    }
+
+    override suspend fun getForegroundInfo(): ForegroundInfo {
+
+        notificationChannelsManager.createRegularChannel(
+            NotificationConstants.OTHER_CHANNEL_ID,
+            NotificationConstants.OTHER_CHANNEL_NAME
+        )
+
+        val notification = NotificationCompat.Builder(applicationContext, NotificationConstants.OTHER_CHANNEL_ID)
+            .setSmallIcon(R.drawable.notification_icon_small)
+            .setAutoCancel(true)
+            .setSilent(true)
+            .setCategory(NotificationCompat.CATEGORY_SERVICE)
+            .setProgress(0, 0, true)
+            .setContentTitle(applicationContext.getString(R.string.migration_title))
+            .setContentText(applicationContext.getString(R.string.migration_message))
+            .setPriority(NotificationCompat.PRIORITY_MIN)
+            .setContentIntent(openAppPendingIntent(applicationContext))
+            .build()
+
+        return ForegroundInfo(NotificationConstants.SINGLE_USER_MIGRATION_NOTIFICATION_ID, notification)
+    }
+
+    companion object {
+        const val NAME = "single_user_migration"
+        const val USER_ID_INPUT_DATA = "single_user_migration_user_id"
+    }
+}
+
+suspend fun WorkManager.enqueueSingleUserMigrationWorker(userId: UserId): Flow<MigrationData> {
+    val request = OneTimeWorkRequestBuilder<SingleUserMigrationWorker>()
+        .setExpedited(OutOfQuotaPolicy.RUN_AS_NON_EXPEDITED_WORK_REQUEST)
+        .setBackoffCriteria(BackoffPolicy.EXPONENTIAL, MIN_BACKOFF_MILLIS, TimeUnit.MILLISECONDS)
+        .setInputData(workDataOf(SingleUserMigrationWorker.USER_ID_INPUT_DATA to userId.toString()))
+        .setConstraints(
+            Constraints.Builder()
+                .setRequiredNetworkType(NetworkType.CONNECTED)
```

Comment:
Do we need network to migrate conversations and messages?

Reply:
nope, i will remove this one